tidy3d/components/geometry.py

```diff
@@ -3280,8 +3289,374 @@ def _surface_area(self, bounds: Bound) -> float:
+        ...,
+        title="Surface mesh data",
+        description="Surface mesh data.",
+    )
+
+    @pydantic.root_validator(pre=True)
+    def _check_trimesh_library(cls, values):
+        """Check if the trimesh package is imported."""
+        if not TRIMESH_AVAILABLE:
+            raise ImportError(
+                "The package 'trimesh' was not found. Please install the 'surfacemesh' "
+                "dependencies to use 'CustomSurfaceMeshGeometry'. For example: "
+                "pip install 'tidy3d[surfacemesh]'."
+            )
+        return values
+
+    @pydantic.validator("mesh_dataset", pre=True, always=True)
+    def _warn_if_none(cls, val: SurfaceMeshDataset) -> SurfaceMeshDataset:
+        """Warn if the Dataset fails to load."""
+        if isinstance(val, dict):
+            if any((v in DATA_ARRAY_MAP for _, v in val.items() if isinstance(v, str))):
+                log.warning("Loading 'mesh_dataset' without data.")
+                return None
+        return val
+
+    @pydantic.validator("mesh_dataset", always=True)
+    def _check_mesh(cls, val: SurfaceMeshDataset) -> SurfaceMeshDataset:
+        """Check that the mesh is valid."""
+        if val is None:
+            return None
+        mesh = cls._triangles_to_trimesh(val.surface_mesh)
+        if not mesh.is_watertight:
+            raise ValidationError(
+                "The provided mesh is not watertight. "
+                "You can see which faces are broken using 'trimesh.repair.broken_faces'."
+            )
+        if not mesh.is_winding_consistent:
+            raise ValidationError(
+                "The provided mesh does not have consistent winding. "
+                "Consider using 'trimesh.repair.fix_winding'."
+            )
+        if not mesh.is_volume:
+            raise ValidationError("The provided mesh does not represent a valid (finite) volume.")
+        if not all(np.array(mesh.area_faces) > fp_eps):
+            raise ValidationError("The provided mesh has triangles with zero area.")
+        return val
+
+    @classmethod
+    def from_stl(
+        cls,
+        filename: str,
+        scale: float = 1.0,
+        origin: Tuple[float, float, float] = (0, 0, 0),
+        solid_index: int = None,
+        **kwargs,
+    ) -> Union[CustomSurfaceMeshGeometry, GeometryGroup]:
+        """Load a :class:`.CustomSurfaceMeshGeometry` directly from an STL file.
+        The ``solid_index`` parameter can be used to select a single solid from the file.
+        Otherwise, if the file contains a single solid, it will be loaded as a
+        :class:`.CustomSurfaceMeshGeometry`; if the file contains multiple solids,
+        they will all be loaded as a :class:`.GeometryGroup`.
+
+        Parameters
+        ----------
+        filename : str
+            The name of the STL file containing the surface geometry mesh data..
+        scale : float
+            The length scale for the loaded geometry (um).
```

Comment:
so if scale = 10, does that mean that a vertex of (1,0,0) is at x=10um in tidy3d units? Maybe instead of `(um)` we should clarify this a bit. For example, add another sentence explaining how a supplied data point is interpreted in tidy3d um as a function of scale.